yarn-project/node-keystore/src/loader.ts

```diff
@@ -230,8 +236,31 @@ export function mergeKeystores(keystores: KeyStore[]): KeyStore {
           }
           attesterAddresses.add(key);
         }
+
+        // When merging v1 validators into a v2+ result, preserve original fallback behavior
+        // by explicitly setting publisher/coinbase/feeRecipient if they're missing
+        if (keystore.schemaVersion === 1 && schemaVersion === 2) {
```

Comment:
This feels like it is going to get ugly very easily. Why do we need to do this?

I think if somebody tries to merge 2 keystores with different versions we should just throw.